src/datapoints-graph/charts/echarts-options.service.ts

```diff
@@ -135,10 +135,15 @@ export class EchartsOptionsService {
     renderType: Exclude<DatapointChartRenderType, 'area'>,
     idx: number,
     isMinMaxChart = false
-  ): SeriesOption & { datapointId: string; datapointLabel: string } {
+  ): SeriesOption & {
+    datapointId: string;
+    datapointLabel: string;
+    datapointUnit: string;
+  } {
```

Comment:
I suggest to extract type 
```
{
    datapointId: string;
    datapointLabel: string;
    datapointUnit: string;
  } 
  ```
  to separate type/ interface and call it something like SeriesDatapointInfo (keep in mind that datapointId is also used in chart-realtime.service.ts as SeriesOption & { datapointId: string } so if you change it here, please align types in realtime service too)

Reply:
Changed this... but I wasn't sure if I should still use a union type or switch to an interface that extends SeriesOption. I choose the union type out of simplicity.  Any opinion?

Reply:
you mean **intersection** type (SeriesOption **&** SeriesDatapointInfo) ? I think both ways are just fine